Jenkinsfile

```diff
@@ -34,98 +49,95 @@ pipeline {
-                        sh "npm run zip"
-                        sh "npm run docs"
-                        sh "aws s3 cp ./res/provider ${S3_LOC}/ --recursive"
-                        sh "aws s3 cp ./dist/provider ${S3_LOC}/ --recursive"
-                        sh "aws s3 cp ./dist/client/openfin-" + SERVICE_NAME + ".js ${S3_LOC}/"
-
-                        sh "aws s3 cp ./dist/docs ${DOCS_CHANNEL_LOC} --recursive"
-                        sh "aws s3 cp ./dist/docs ${DOCS_VERSIONED_LOC} --recursive"
-                        sh "aws s3 cp ./dist/provider/app.json ${MANIFEST_LOC}"
-                    }
-                }
-                stage("Deploy Client to NPM (Production)") {
-                    input {
-                        message "Would you like to deploy the client to NPM?"
-                        parameters {
-                            choice(name: 'DEPLOY_CLIENT', choices: ['Yes', 'No'], description: '') 
-                        }
-                    }
-                    when {expression { DEPLOY_CLIENT == 'Yes' }}
-                    steps {
-                        withCredentials([string(credentialsId: "NPM_TOKEN_WRITE", variable: 'NPM_TOKEN')]) {
-                            sh "echo //registry.npmjs.org/:_authToken=$NPM_TOKEN > $WORKSPACE/.npmrc"
-                        }
-                        echo "publishing to npm, version: " + BUILD_VERSION
-                        sh "npm publish"
-                    }
-                }
+            when { anyOf { branch 'develop' ; branch 'master' } }
+            steps {
+                deployToS3()
+                deployToNPM()
             }
         }
     }
 }
+
+def configure() {
+    def config = readJSON file: './services.config.json'
+    GIT_SHORT_SHA = sh ( script: "git rev-parse --short HEAD", returnStdout: true ).trim()
+    PKG_VERSION = sh ( script: "node -pe \"require('./package.json').version\"", returnStdout: true ).trim()
+    SERVICE_NAME = config.SERVICE_NAME
+
+    if (env.BRANCH_NAME == 'master') {
+        BUILD_VERSION = PKG_VERSION
+        CHANNEL = "stable"
+        MANIFEST_NAME = "app.json"
+    } else {
+        BUILD_VERSION = PKG_VERSION + "-alpha." + env.BUILD_NUMBER
+        CHANNEL = "staging"
+        MANIFEST_NAME = "app.staging.json"
+    }
+
+    DIR_BUILD_ROOT = env.DSERVICE_S3_ROOT + SERVICE_NAME + "/"
+    DIR_BUILD_VERSION = DIR_BUILD_ROOT + BUILD_VERSION
+
+    DIR_DOCS_ROOT = env.DSERVICE_S3_ROOT_DOCS + SERVICE_NAME + "/"
+    DIR_DOCS_CHANNEL = DIR_DOCS_ROOT + CHANNEL
+    DIR_DOCS_VERSION = DIR_DOCS_ROOT + BUILD_VERSION
+}
+
+def buildProject() {
+    sh "npm run clean"
+    sh "SERVICE_VERSION=${BUILD_VERSION} npm run build"
+    sh "echo ${GIT_SHORT_SHA} > ./dist/SHA.txt"
+
+    sh "npm run zip"
+    sh "npm install bootprint@2.0.1 bootprint-json-schema@2.0.0-rc.3 --no-save"
+    sh "npm run docs"
+}
+
+def addReleaseChannels() {
+    if (env.BRANCH_NAME == 'master') {
+        sh "npm run channels"
```

Comment:
I think we're missing the `channels` declaration in `package.json`. Unless we get it automatically from `svc-tools`?

Reply:
My bad - this went through a few revisions, ended up with the script missing. Have added to `package.json` now.